src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -79,26 +133,13 @@ Status CoreWorkerPlasmaStoreProvider::Get(
       should_break = remaining_timeout <= 0;
     }
 
-    std::vector<std::shared_ptr<RayObject>> batch_results;
-    RAY_RETURN_NOT_OK(
-        local_store_provider_.Get(batch_ids, batch_timeout, task_id, &batch_results));
-
-    // Add successfully retrieved objects to the result list and remove them from
-    // remaining.
-    uint64_t successes = 0;
-    for (size_t i = 0; i < batch_results.size(); i++) {
-      if (batch_results[i] != nullptr) {
-        successes++;
-        const auto &object_id = batch_ids[i];
-        (*results)[remaining[object_id]] = batch_results[i];
-        remaining.erase(object_id);
-        if (IsException(*batch_results[i])) {
-          should_break = true;
-        }
-      }
-    }
+    size_t previous_size = remaining.size();
+    RAY_RETURN_NOT_OK(FetchAndGetFromPlasmaStore(remaining, batch_ids, batch_timeout,
+                                                 /*fetch_only=*/false, task_id, results,
+                                                 &got_exception));
+    should_break = should_break || got_exception;
 
-    if (successes < batch_ids.size()) {
+    if ((remaining.size() - previous_size) < batch_ids.size()) {
```

Comment:
this doesn't look right, as `previous_size` should be no less than `remaining.size()`

it should be `previous_size - remaining.size()` ?